### PR TITLE
Add documentation for inlets-operator

### PR DESCRIPTION
**Pull request by utsavanand2**
Signed-off-by: Utsav Anand <utsavanand2@gmail.com>

Adds the documentation for installing inlets-operator across DigitalOcean, EC2, GCE, Linode, and Azure
~**(EC2 is to be added yet)**~

### CODE REVIEW

**Comment by utsavanand2** [comment]
Not Spam

**Comment by alexellis** [comment]
Thanks for the PR, I'm going to turn Derek off for this org as we've not been bothered by Hacktoberfest spam :)

**Comment by alexellis** [comment]
Why is EC2 yet to be added?

**Comment by alexellis** [comment]
Does it just not exist in the upstream readme yet? I added steps for a service accounts the docs for inletsctl, which you can copy and adapt if you like?

**Comment by alexellis** [comment]
https://docs.inlets.dev/#/tools/inletsctl?id=example-usage-with-aws-ec2

File: docs/tools/inlets-operator.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+inlets-operator adds public LoadBalancers to your local Kubernetes clusters.
+
+View the code on GitHub: [inlets/inletsctl](https://github.com/inlets/inletsctl)
+
+## Using DigitalOcean for the exit node
+
+> Note: this example is now multi-arch, so it's valid for `x86_64`, `ARMHF`, and `ARM64`.
+
+You can also run the operator in-cluster, a ClusterRole is used since Services can be created in any namespace, and may need a tunnel. Install the inlets-operator using a single command with [arkade](https://get-arkade.dev/). arkade runs against any Kubernetes cluster and wraps the helm command-line.
```
Comment (alexellis) [inline docs/tools/inlets-operator.md:11]: Remove `You can also run the operator in-cluster, `

File: docs/tools/inlets-operator.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
+# inlets-operator reference documentation
+
+inlets-operator adds public LoadBalancers to your local Kubernetes clusters.
+
+View the code on GitHub: [inlets/inletsctl](https://github.com/inlets/inletsctl)
+
+## Using DigitalOcean for the exit node
```
Comment (alexellis) [inline docs/tools/inlets-operator.md:7]: Can we change the titles to:

"Create exit-servers on DigitalOcean" and so forth?

File: docs/tools/inlets-operator.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+# inlets-operator reference documentation
+
+inlets-operator adds public LoadBalancers to your local Kubernetes clusters.
+
+View the code on GitHub: [inlets/inletsctl](https://github.com/inlets/inletsctl)
```
Comment (alexellis) [inline docs/tools/inlets-operator.md:5]: I would link to inlets-operator, not inletsctl

File: docs/tools/inlets-operator.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
+# inlets-operator reference documentation
+
+inlets-operator adds public LoadBalancers to your local Kubernetes clusters.
```
Comment (alexellis) [inline docs/tools/inlets-operator.md:3]: Please add one generic statement and example at the top showing inlets-operator installed with helm, and then one with arkade.

File: docs/tools/inlets-operator.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+```bash
+kubectl apply -f ./artifacts/crds/
+kubectl create secret generic inlets-access-key --from-file=inlets-access-key=~/Downloads/client_credentials.json
+helm repo add inlets https://inlets.github.io/inlets-operator/
+helm repo update
+helm upgrade inlets-operator --install inlets/inlets-operator \
+  --set provider=azure,region=eastus,subscriptionID=<Azure Subscription ID> 
+```
+
+## Using a provider which requires an Access Key and Secret Key? (AWS EC2, Scaleway)
```
Comment (alexellis) [inline docs/tools/inlets-operator.md:147]: Should we just break these out into specific sections if they do not exist already?

Comment (utsavanand2) [reply docs/tools/inlets-operator.md:233]: Yes it is there in the ec2 example because ec2 needs the `secret-key-file`, so this section can be omitted.

**Comment by utsavanand2** [comment]
@alexellis Does it look any better?

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
Now just add your link in the main README on the homepage, then when it's merged, remove all the duplicated text from the inlets-operator README and link to here.

Thanks for this 🍻

**Comment by utsavanand2** [comment]
@alexellis Happy to help!

**Comment by utsavanand2** [comment]
@alexellis I've resolved the merge conflicts.

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
Merged. Please can you do the next part in the inlets-operator README?

**Pull request merged**